advanced/wallets/react-wallet-v2/src/lib/Bip122Lib.ts

```diff
@@ -0,0 +1,158 @@
+      pubkey: restored.publicKey,
+      network: bitcoin.networks.testnet
+    }).address!
+    this.account = restored
+    this.mnemonic = mnemonic
+    this.wif = restored.toWIF()
+    console.log('btc address', this.address)
+    // console.log('key', bitcoin)
+    // const keyPair = key ? ECPair.fromWIF(key) : ECPair.makeRandom()
+    // this.privateKey = keyPair.toWIF()
+    // const { address } = bitcoin.payments.p2pkh({
+    //   pubkey: keyPair.publicKey,
+    //   network: bitcoin.networks.testnet
+    // })
+    // this.address = address!
+
+    // console.log('btc address', address)
+  }
+
+  static async init({ privateKey }: IInitArguments) {
+    return new Bip122Lib(privateKey)
+  }
+
+  public getAddress() {
+    return this.address
+  }
+
+  public getPrivateKey() {
+    return this.mnemonic
+  }
+
+  public signMessage(message: string) {
+    var keyPair = ECPair.fromWIF(this.wif)
+    var privateKey = keyPair.privateKey!
+    var signature = bitcoinMessage.sign(message, privateKey, keyPair.compressed, {
+      segwitType: 'p2wpkh'
+    })
+    console.log('sig', signature.toString('base64'))
+    console.log('isValid', bitcoinMessage.verify(message, this.address, signature, undefined, true))
+    return {
+      signature: signature.toString('base64'),
+      segwitType: 'p2wpkh'
+    }
+  }
+
+  public async signTransaction(params: {
+    address: string
+    amount: number
+    transactionType: string
+  }) {
+    const { address, amount, transactionType } = params
+    if (this.address !== address) {
+      throw new Error(`Unkown address: ${address}`)
+    }
+
+    if (amount < 0) {
+      throw new Error(`Invalid amount: ${amount}`)
+    }
+
+    if (transactionType !== 'p2wpkh') {
+      throw new Error(`Only p2wpkh transactions are supported. Received: ${transactionType}`)
+    }
+
+    const payment = bitcoin.payments.p2wpkh({
+      pubkey: this.account.publicKey
+    })
+
+    const utxos = await this.getUtXOs(this.address)
+    if (!utxos || utxos.length === 0) {
+      throw new Error(`No UTXOs found for address: ${this.address}`)
+    }
+    console.log('utxos', utxos)
+    const availableBalance = this.getAvailableBalance(utxos)
+    console.log('availableBalance', availableBalance)
+
+    if (availableBalance < amount) {
+      throw new Error(`Insufficient funds: ${availableBalance}`)
+    }
+
+    const amountToTransfer = Math.ceil((amount / 100) * 98) // 2% validators fee
+    console.log('amountToTransfer', amountToTransfer)
+    console.log('output', payment.output?.toString('hex'))
+    const psbt = new bitcoin.Psbt({ network: bitcoin.networks.testnet })
+
+    const scriptHash = Buffer.from(payment.output?.toString('hex')!, 'hex')
+    for (const utxo of utxos) {
+      psbt.addInput({
+        hash: utxo.txid,
+        index: utxo.vout,
+        witnessUtxo: {
+          script: scriptHash,
+          value: utxo.value
+        }
+      })
+    }
+    psbt.addOutput({
+      address: this.getAddress(),
```

Comment:
Are we supposed to transfer to `address` instead of `this.getAddress`? `this.getAddress` would be the transaction change which we would need to handle anyways I think

Reply:
if not a blocker as first iteration we can go with single address and at later point begin deriving new ones on a transaction, wdyt?